### PR TITLE
update history for 2.0 release

### PR DESCRIPTION
**Pull request by tpike3**
With requested corrections, release date 15 July. Feel free to make any additional corrections to the history. --- supersedes #1727, this merges into main branch not v2.0

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,62 @@
...
+* space: raise exception when pos is out of bounds in `Grid.get_neighborhood` #1524
+* space: remove deprecations (#1520, #1687, #1688):
+    * `find_empty()`: convert this to `move_to_empty()`
+    * `num_agents`: removed parameter from `move_to_empty()`
+    * `position_agent()`: convert this to `place_agent`
+    * `neighbor_iter()`: convert this to `iter_neighborhood()`
+* batchrunner: remove deprecations #1627
+    * `class BatchRunner` and `class BatchRunnerMP`: convert these to `batch_run()`
+* visualization: easier visualization creation #1693
+    * `UserSettableParameter(['number', 'slider','checkbox', 'choice', `StaticText`]): convert to `NumberInput`,`Slider`, `CheckBox`, `Choice`, `StaticText`
```
Comment (rht) [inline HISTORY.rst:26]: Can you add an example by linking to https://github.com/projectmesa/mesa-examples/blob/db2ec0383eb3b1868e91c828101e84cce97bbb63/examples/boltzmann_wealth_model/boltzmann_wealth_model/server.py#L25-L32. I want to make sure people don't have problem migrating.

File: HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,62 @@
...
+* space: change `coord_iter` to return `(content,(x,y))` instead of `(content, x,y)`; this reduces known errors of scheduler to grid mismatch #1566, #1723
+* space: change NetworkGrid `get_neighbors` to `get_neighborhood`; improves performance #1542
+* space: raise exception when pos is out of bounds in `Grid.get_neighborhood` #1524
+* space: remove deprecations (#1520, #1687, #1688):
+    * `find_empty()`: convert this to `move_to_empty()`
+    * `num_agents`: removed parameter from `move_to_empty()`
+    * `position_agent()`: convert this to `place_agent`
+    * `neighbor_iter()`: convert this to `iter_neighborhood()`
+* batchrunner: remove deprecations #1627
+    * `class BatchRunner` and `class BatchRunnerMP`: convert these to `batch_run()`
```
Comment (rht) [inline HISTORY.rst:24]: Add example: https://github.com/projectmesa/mesa-examples/blob/db2ec0383eb3b1868e91c828101e84cce97bbb63/examples/bank_reserves/batch_run.py#L188-L221

File: HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,62 @@
...
+    * `UserSettableParameter(['number', 'slider','checkbox', 'choice', `StaticText`]): convert to `NumberInput`,`Slider`, `CheckBox`, `Choice`, `StaticText`
+
+
+**New Features:**
+
+* datacollector: can now handle data collection by agent type #1419, #1702
+* time: allows for model level `StageActivation` #1709
+* visualization: `ChartModule` can have dynamically named properties #1685
+* visualization: improved stop server to end visualizations #1646
+* *experimental* python front end option: integrated the initial prototype of the pure python front end option #1698, #1726
```
Comment (rht) [inline HISTORY.rst:35]: This needs to be emphasized in the early section of the release note. Anyone reading the release note would have missed this sentence very easily. You could say that this is the main change of 2.0, other than the breaking changes.

Also, needs to say early on that we are switching to the pure Python viz as the default, if it goes well.

File: HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,71 @@
...
+2.0.0 (2023-04-15) Wellton
+++++++++++++++++++++++++++
+
+**Special notes**
+
+Mesa 2.0 includes:
+    * **an experimental pure python user interface/ visualization that is also jupyter compatible please see the** `visualization tutorial`_
+    * several breaking changes that provide significant improvements to Mesa.
+
+.. _visualization tutorial: https://mesa.readthedocs.io/
```
Comment (rht) [inline HISTORY.rst:15]: Last comment: needs to link to the actual visualization tutorial.

**Review by rht — APPROVED** [review]

**Comment by rht** [comment]
I usually bump the version in the same commit where the HISTORY.rst was updated. See:
- https://github.com/projectmesa/mesa/commit/9a07a48526f11b78e462a9bab390366c95443814
- https://github.com/projectmesa/mesa/commit/976ddfc8a1e5feaaf8007a7abaa9abc7093881a0

**Comment by tpike3** [comment]
> I usually bump the version in the same commit where the HISTORY.rst was updated. See:
> 
> * [9a07a48](https://github.com/projectmesa/mesa/commit/9a07a48526f11b78e462a9bab390366c95443814)
> * [976ddfc](https://github.com/projectmesa/mesa/commit/976ddfc8a1e5feaaf8007a7abaa9abc7093881a0)

done

**Comment by EwoutH** [comment]
Congratulations on the 2.0 release! Looks like a serious clean-up, and looking forward to testing and using the data collection by agent type and the improved visualization.

Thanks for all the hard work!

**Comment by rht** [comment]
The visualization is slightly slower because it is a Matplotlib redraw every time it gets updated, but it is fast enough for small scale models. The visualization is mainly for small scale exploration anyway. In return, it is much simpler to customize.

**Comment by rht** [comment]
The version on PyPI is not yet 2.0.

**Pull request merged**